miru/button.py

```diff
@@ -145,30 +142,24 @@ def _from_component(cls, component: hikari.PartialComponent, row: t.Optional[int
         )
 
     def _build(self, action_row: hikari.api.MessageActionRowBuilder) -> None:
-        button: t.Union[
-            hikari.api.InteractiveButtonBuilder[hikari.api.MessageActionRowBuilder],
-            hikari.api.LinkButtonBuilder[hikari.api.MessageActionRowBuilder],
-        ]
-        if self.emoji is None and self.label is None:
-            raise TypeError("Must provide at least one of 'emoji' or 'label' when building Button.")
-
-        if self.url is not None:
-            button = action_row.add_button(hikari.ButtonStyle.LINK, self.url)
-        else:
-            button = action_row.add_button(self.style, self.custom_id)
-
+        config = {}
         if self.label:
-            button.set_label(self.label)
+            config["label"] = self.label
         if self.emoji:
-            button.set_emoji(self.emoji)
-        button.set_is_disabled(self.disabled)
+            config["emoji"] = self.emoji
 
-        button.add_to_container()
+        config["is_disabled"] = self.disabled
+
+        if self.url is not None:
+            action_row.add_link_button(self.url, **config)
+        else:
+            action_row.add_interactive_button(self.style, self.custom_id, **config)
 
 
 def button(
     *,
     label: t.Optional[str] = None,
+    url: t.Optional[str] = None,
```

Comment:
This is a decorator, and you wouldn't use it to decorate a function that has no callback (i.e. a link button)

Reply:
I think even if it doesn't have a callback, it's good that to have things generalized, meaning the ability to make all kinds of buttons in the decorator.

```py
@miru.button(label="Click me!",url="google.com") #this will be url button since we pass url parameter
async def basic_url_button(self, button: miru.Button, ctx: miru.ViewContext) -> None:
      pass
   
@miru.button(label="Click me!", style=hikari.ButtonStyle.SUCCESS)
async def basic_button(self, button: miru.Button, ctx: miru.ViewContext) -> None:
    await ctx.respond("You clicked me!")
    @miru.text_select(
        placeholder="Select me!",
        options=[
            miru.SelectOption(label="Option 1"),
            miru.SelectOption(label="Option 2"),
        ],
    )
    
async def basic_select(self, select: miru.TextSelect, ctx: miru.ViewContext) -> None:
    await ctx.respond(f"You've chosen {select.values[0]}!")
```

or, the other approach is to create another decorator called '@miru.url_button' that takes a 'url' parameter

Reply:
do you mean this?
```py
button = miru.URLButton(url="google.com", label="google")

@miru.button(label="Click me!", style=hikari.ButtonStyle.SUCCESS)
async def basic_button(self, button: miru.Button, ctx: miru.ViewContext) -> None:
    await ctx.respond("You clicked me!")
    @miru.text_select(
        placeholder="Select me!",
        options=[
            miru.SelectOption(label="Option 1"),
            miru.SelectOption(label="Option 2"),
        ],
    )
    
async def basic_select(self, select: miru.TextSelect, ctx: miru.ViewContext) -> None:
    await ctx.respond(f"You've chosen {select.values[0]}!")
```

seems this will ruin the pattern :)

Reply:
I don't think adding empty callbacks that do nothing is a good idea frankly, it gives the perception that one could execute logic in a place where you actually can't.